app/controllers/planning_applications_controller.rb

```diff
@@ -34,6 +36,39 @@ def index
 
   def show; end
 
+  def new
+    @planning_application = PlanningApplication.new
+  end
+
+  def edit; end
+
+  def create
+    @planning_application = PlanningApplication.new(planning_application_params)
+    @planning_application.assign_attributes(local_authority: current_local_authority)
+
+    if @planning_application.save
+      audit("created", nil, current_user.name)
+      flash[:notice] = "Planning application was successfully created."
+      redirect_to planning_application_documents_path(@planning_application)
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @planning_application.update(planning_application_params)
+      planning_application_params.keys.map do |p|
+        if @planning_application.saved_change_to_attribute?(p)
+          audit("updated", "Changed from: #{@planning_application.saved_changes[p][0]} \r\n Changed to: #{@planning_application.saved_changes[p][1]}", p.split("_").join(" ").capitalize)
+        end
+      end
+      flash[:notice] = "Planning application was successfully updated."
+      redirect_to planning_application_url
```

Comment:
huh, how is this working without passing the @planning_application to this method? I'd have thought this gave an error, but I guess it's not? Shouldn't it be `redirect_to planning_application_url(@planning_application)`

Reply:
this one not only works  - it also doesn't create that problem with brakeman unlike what happens everytime we use @planning_application - brakeman doesn't like the @planning_application id params for some reason.

it seems that rails routes has a neat little feature where the id is filled from the action:

https://stackoverflow.com/questions/40785322/why-redirecting-to-show-action-without-passing-the-id-param-work-in-rails

Reply:
huh! I learned a thing.